tools/killswitch/src/output.rs

```diff
@@ -0,0 +1,209 @@
+use serde_json::{json, Value};
+use std::collections::HashMap;
+
+/// KillSwitch response showing success / failure of configuration
+/// and tx submission. Gets serialized to json
+#[derive(Serialize)]
+pub(crate) struct Output {
+    /// The original command `killswitch` was run with
+    pub command: String,
+    /// The success / failure message
+    pub message: Message,
+}
+
+/// A wrapper for success / failure messages
+#[derive(Serialize)]
+#[serde(untagged)]
+pub(crate) enum Message {
+    /// An wrapper for a single error we bailed on
+    SimpleError(String),
+    /// A full results message as a json `Value`
+    FullMessage(Value),
+}
+
+impl From<Error> for Message {
+    /// Convert a blocking `Error` to `Message`
+    fn from(error: Error) -> Self {
+        Message::SimpleError(format!("{:?}", error))
+    }
+}
+
+/// Build output `Message::FullMessage(Value)` accepting a set
+/// of errored channels as well as successful channels
+#[allow(clippy::type_complexity)]
+pub(crate) fn build_output_message(
+    bad: Vec<(Channel, Vec<Error>)>,
+    good: Vec<(Channel, TxOutcome)>,
+) -> Message {
+    let mut replicas = bad
+        .into_iter()
+        .map(|(channel, errors)| {
+            let val = json!({
+                "result": {
+                    "status": "error",
+                    "tx_hash": serde_json::Value::Null,
+                    "message": errors
+                        .iter()
+                        .map(|e| format!("{}", e))
+                        .collect::<Vec<String>>(),
+                }
+            });
+            (channel.clone(), (false, (channel.replica, val)))
+        })
+        .collect::<Vec<(_, (_, _))>>();
+    replicas.extend(good.into_iter().map(|(channel, tx)| {
+        let val = json!({
+            "result": {
+                "status": "success",
+                "tx_hash": format!("{:?}", tx.txid),
+                "message": serde_json::Value::Null,
+            }
+        });
+        (channel.clone(), (true, (channel.replica, val)))
+    }));
+    let mut homes: HashMap<String, Vec<(bool, (String, Value))>> = HashMap::new();
+    for (channel, (success, replica)) in replicas {
+        if let Some(replicas) = homes.get_mut(&channel.home) {
+            replicas.push((success, replica));
+        } else {
+            homes.insert(channel.home, vec![(success, replica)]);
+        }
+    }
+    Message::FullMessage(json!({
+        "homes": homes.into_iter().map(|(home, replicas)| {
+            // report error for *any* errors encountered
+            let success = replicas.iter().all(|(s, _)| *s);
+            (home, json!({
+                "status": if success { "success" } else { "error" },
+                "message": {
+                    "replicas": replicas
+                        .into_iter()
+                        .map(|(_, replica)| replica)
+                        .collect::<HashMap<String, Value>>(),
+                }
+            }))
```

Comment:
Likely easier if we made this JSON message into concrete type and derived `serde::{Serialize, Deserialize}`. Think that would be much cleaner and more readable making this stringly typed